### ISSUE TITLE
[BUG] Java patterns with * in the middle are not supported

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#481]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0

### Priority

Major

### Current Behavior

The following rule:
```yaml
  ruleID: hibernate-search-00960
  when:
    or:
    - java.referenced:
        location: PACKAGE
        pattern: org.hibernate.search*Facet*
    - java.referenced:
        location: PACKAGE
        pattern: org.hibernate.search*facet*
```
does not get triggered because of asterisks in the middle.

### Expected Behavior

The rule should trigger against, for instance:
```java
import org.hibernate.search.annotations.Facet;
```

### How Reproducible

Always (Default)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by jmle** [comment]
The rule above works when written as:
```yaml
  ruleID: hibernate-search-00960
  when:
    or:
    - java.referenced:
        location: PACKAGE
        pattern: org.hibernate.search.*.Facet*
    - java.referenced:
        location: PACKAGE
        pattern: org.hibernate.search.*.facet*
```
This won't be valid for all cases of `*` happening in the middle of patterns though.

**Comment by shawn-hurley** [comment]
@jmle Can you add more on the cases that don't work, wondering if we want to solve those as well?

**Comment by pranavgaikwad** [comment]
@jmle this should be fixed via https://github.com/konveyor/rulesets/pull/67 are you good to close this?
